File: ScpProject/BL/BusinessObjects/Survey/Question.cs
using System.Collections.Generic;
using Models.Enums;

namespace BL.BusinessObjects.Survey
{
    public class QuestionDTO
    {
        public int SurveysToQuestionsId { get; set; }
        public int QuestionId { get; set; }
        public string Question { get; set; }
        public QuestionTypeEnum QuestionType { get; set; }
        public int SurveyId { get; set; }
        public List<NewYesNoThreshold> YesNoThresholds { get; set; }
        public List<NewScaleThreshold> ScaleThresholds { get; set; }
        public bool CanModify { get; set; }
    }
}

File: ScpProject/StrengthConApp/src/app/Models/Question.ts
import { ScaleThreshold } from '../Models/Survey/ScaleThreshold';
import { YesNoThreshold } from '../Models/Survey/YesNoThreshold'
export class Question {
    public QuestionId: number;
    public Question: string;
    public QuestionType: QuestionType;
    public SurveyId: number;
    public YesNoThresholds: YesNoThreshold[] = [];
    public ScaleThresholds: ScaleThreshold[] = [];
    public CanModify : boolean;
}

export class MappedQuestionToSurvey {
    public SurveyToQuestionId: number;
    public QuestionId: number;
    public Question: string;
    public QuestionType: QuestionType;
    public SurveyId: number;
    public YesNoThresholds: YesNoThreshold[] = [];
    public ScaleThresholds: ScaleThreshold[] = []
    public CanModify : boolean;
}


export enum QuestionType {
    Boolean = 1,
    Scale = 2,
    OpenEnded = 3
}